src/server/hook_func.c

```diff
@@ -6751,6 +6944,482 @@ get_hook_rescdef_checksum(void)
+ * @param[in,out] 	accept_flag -  return 1 if event accept flag is true.
+ * @param[in,out] 	reject_flag -  return 1 if event reject flag is true.
+ * @param[in,out] 	reject_msg -  the reject message if reject_flag is 1.
+ * @param[in]		reject_msg_size -  size of reject_msg buffer.
+ * @param[in,out] 	pjob -  job in question, where if present (not NULL),
+ *			        it gets filled in with the
+ *				"pbs.event().job" entries in 'input_file'.
+ *				'pjob' can be NULL in periodic hooks, since
+ *				 periodic hooks are not tied to jobs.
+ *				 Note that pbs.event().job_list[<jobid>] entries
+ *				 in 'input_file' fill in the individual
+ *				 <jobid>'s job struct entry in the system, and
+ *				 not the passed 'pjob' structure.
+ * @param[in]		phook -  hook that executed of which we're getting the
+ *				results. If non-NULL, then phook->user is
+ *				used to validate 'pbs.event().job.euser' line
+ *				in 'input_file'.
+ *				If main Mom is reading a job related hook
+ *				results file, phook will be null; an entry in
+ *				the file should give us the hook name from which
+ *				phook is found.
+ * @param[out]		hook_output - struct of parameters to fill in output.
+ *
+ * @return int
+ * @retval	0 for success
+ * @retval	non-zero for failure;  the returned parameters (accept_flag,
+ *		reject_flag, svrvnalist and pjob) may be invalid and should be
+ *		ignored.   The list svrvnalist could have mallocate space and
+ *		should be freed by the calling program.
+ */
+int
+get_server_hook_results(char *input_file, int *accept_flag, int *reject_flag, char *reject_msg,
+	int reject_msg_size, job *pjob, hook *phook, hook_output_param_t *hook_output)  {
+
+	char	*name_str;
+	char	*resc_str;
+	char	*obj_name;
+	char	*data_value;
+	char	*vname_str;
+	int	rc = -1;
+	char	*pc, *pc1, *pc2, *pc3, *pc4;
+	char	*in_data = NULL;
+	size_t	ll;
+	FILE	*fp;
+	char	*p;
+	int	vn_obj_len = strlen(EVENT_VNODELIST_OBJECT);
+	int	job_obj_len = strlen(EVENT_JOBLIST_OBJECT);
+	char	hook_job_outfile[MAXPATHLEN + 1];
+	FILE	*fp2 = NULL;
+	char	*line_data = NULL;
+	int	line_data_sz;
+	long int endpos;
+	char	hook_euser[PBS_MAXUSER + 1];
+	int	arg_list_entries = 0;
+	int	b_triple_quotes = 0;
+	int	e_triple_quotes = 0;
+	char	buf_data[STRBUF];
+	int	buf_data_sz = STRBUF;
+	int	valln = 0;
+	svrattrl *plist = NULL;
+	struct pbsnode *pnode;
+	int	bad = 0;
+	int	ndtype_flag = 0;
+	char	*pbse_err;
+	char	raw_err[10];
+
+	/* Preset hook_euser for later.  If we are reading a job related     */
+	/* copy of hook results, there will be one or more (one per hook)    */
+	/* pbs_event().hook_euser=<value> entries.  In that case, hook_euser */
+	/* is reset to the <value>.  A null string <value> means PBSADMIN.   */
+	hook_euser[0] = '\0';
+	if (phook && pjob &&  (phook->user == HOOK_PBSUSER)) {
+		strncpy(hook_euser,
+			pjob->ji_wattr[(int)JOB_ATR_euser].at_val.at_str,
+			PBS_MAXUSER);
+	}
+
+	if ((input_file != NULL) && (*input_file != '\0')) {
```

Comment:
Can you please comment/quote an example of how an input_file will look like. Thank!

Reply:
Done. Added few lines.